ogr/services/github/token_managers/tokman_manager.py

```diff
@@ -0,0 +1,43 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+
+import requests
+
+
+import github
+
+
+from ogr.abstract import GithubTokenManager
+from ogr.exceptions import OgrException
+
+
+class TokmanGithubTokenManager(GithubTokenManager):
+    def __init__(self, instance_url: str):
+        self.__instance_url = instance_url
+
+        try:
+            # check if running correctly
+            health = requests.get(f"{self.__instance_url}/api/health")
```

Comment:
Is doing this check here really required?

My concern is that this will create a dependency on the tokman service running at construction time, which is not really required: it should be completely fine to have the tokman service down when the object is created, the only thing that matters is to have it running when one tries to retrieve a token.

In other words, doing this check here, will make the design rigid.

Am I missing something?

Reply:
> Am I missing something?

Nope, GitHub didn't show you latest version apparently or you reviewed commit by commit. Latest version does this check on the first call of `get_pygithub_instance`

Reply:
Yeah, true, I was checking an older commit.

Still, I hold up this comment as valid: the primary use of the `health` API endpoint was to provide a way for OpenShift to check the health of tokman.

I don't see what additional information or benefit do we get here by checking the health before requesting a token.  Whatever error code is found in the `response` should be enough to identify all issues, including the case when the tokman service is not running.

Reply:
Thanks, removed. It was intended as a way to catch problem like typos immediately after running (e.g. service), but since I wanted to enforce laziness, I reorganized it and left it there...